lib/matplotlib/tests/test_legend.py

```diff
@@ -563,3 +563,30 @@ def test_alpha_handles():
         lh.set_alpha(1.0)
     assert lh.get_facecolor()[:-1] == hh[1].get_facecolor()[:-1]
     assert lh.get_edgecolor()[:-1] == hh[1].get_edgecolor()[:-1]
+
+
+def test_warn_big_data_best_loc():
+    fig, ax = plt.subplots()
+    ax.plot(np.arange(200001), label='Is this big data?')
+    with pytest.warns(UserWarning) as records:
+        l = ax.legend()
+        # We need to call an internal set method tricking it into thinking
+        # 'best' location was default.
+        l._set_loc(l.codes['best'], is_initial_setting=True)
```

Comment:
Can the use the private function be circumvented by using an `rc_context`? Maybe this is also not neccesary anymore after my above comment?

Reply:
Probably, I wasn't familiar with that for testing, thanks for the tip!